File: Level1/로또의최고순위와최저순위/solution.py
def solution(lottos, win_nums):
    answer = []
    zero = 0
    win = 0

    for l in lottos:
        if l == 0:
            zero += 1
        elif l in win_nums:
            win += 1

    answer.append(rank(win))
    answer.append(rank(win + zero))
    answer.sort()

    return answer


def rank(num):
    if num == 6:
        return 1
    elif num == 5:
        return 2
    elif num == 4:
        return 3
    elif num == 3:
        return 4
    elif num == 2:
        return 5
    else:
        return 6

File: Level1/로또의최고순위와최저순위/solution.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int num = 0;
    int zero = 0;

    for (int i = 0; i < lottos.size(); i++){
        if (lottos[i] != 0)
            for (int j = 0; j < win_nums.size(); j++){
                if (lottos[i] == win_nums[j])
                    num += 1;
            }
        else
            zero += 1;
    }

    int rank[] = {6, 6, 5, 4, 3, 2, 1};

    answer.push_back(rank[num + zero]);
    answer.push_back(rank[num]);

    return answer;
}